hatch_build.py

```diff
@@ -0,0 +1,41 @@
+"""
+Responsible for defining the custom build hook plugin class.
+"""
+
+from pathlib import Path
+from typing import Any
+import os
+
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+
+
+class CustomBuildHook(BuildHookInterface):
+    """
+    A custom build hook plugin class.
+
+    Subclasses the BuildHookInterface to let know hatchling
+    about a custom build hook plugin and defines the
+    initialize method.
+    """
+
+    def initialize(
+        self,
+        version: str,
+        build_data: dict[str, Any],
+    ) -> None:
+        """
+        Includes the json file in the final build.
+
+        Reads the JSON file path from environment variable and
+        force includes it in the final build.
+        """
+        if not os.environ.get("RUNNING_HATCH_SESSION"):
```

Comment:
Oh I see, you don't want the file generated outside -- here itself is where you should be generating this file / data!